### DISCUSSION TITLE
Checking for presence of a non-null value

### DISCUSSION THREAD

**Discussion by kkarbowiak** [nlohmann/json#3860] [Q&A] [answered]
I have some code that looks like this:
```c++
if (some_json.contains("foo") && !some_json["foo"].is_null()) {
    process(some_json["foo"].get<std::string>());
}
```
I would like to simplify it to something like this:
```c++
if (some_json.contains_non_null("foo")) {
    process(some_json["foo"].get<std::string>());
}
```
Would such usage be common enough to include the `contains_non_null` function in the interface (perhaps with a better name)?

Or is there some already available idiom that I am missing?

**Comment by nlohmann** [comment, marked as answer]
There is currently no other way and also no plan to extend the API. You can define a function yourself though.

**Reply by kkarbowiak** [reply]
Understood. Thanks for the response.

**Reply by gregmarr** [reply]
Note also that your code doesn't guard against `"foo"` being a number or some other non-string type.  You may want to do this instead if you can't guarantee that the JSON you are provided only contains the element as a string, and not as some other type:

    if (some_json.contains("foo") && some_json["foo"].is_string())

**Reply by kkarbowiak** [reply]
Thank you, that's helpful. In my case though the input is after initial validation and the value is of proper type or null.
